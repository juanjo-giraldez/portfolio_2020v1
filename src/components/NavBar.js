import React from 'react';
import { Link, NavLink } from "react-router-dom";
import { useStateBar } from '../hooks/useStateBar';


export const NavBar = () => {

    const [state, open, reset, close] = useStateBar(false)

    return (
        <nav >
            <div >
                <Link to="/" onClick={reset}><svg className="logo" alt="logo"></svg></Link>
            </div>
            <div className="link">
                <div className={state ? "toggle" : "mobile"}>
                    <NavLink className="item" onClick={reset} exact to="/" >Home</NavLink>
                    <NavLink className="item" onClick={reset} exact to="./about" >Sobre m√≠</NavLink>
                    <NavLink className="item" onClick={reset} exact to="./contact" >Contacto</NavLink>
                    <NavLink className="item" onClick={reset} exact to="./skills" >Habilidades</NavLink>
                    <NavLink className="item" onClick={reset} exact to="./proyects"  >Proyectos</NavLink>
                    <div  onClick={close}  ><svg className=" close " alt="close"></svg></div>
                </div>
                <div onClick={open}><svg className="menu" alt="menu"></svg></div>
            </div>
        </nav>
    )
}

