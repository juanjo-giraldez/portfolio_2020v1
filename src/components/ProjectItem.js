
import React from 'react'

export const ProjectItem = ({ demo, id, img, project, technologies, url }) => {

    const result = technologies.map((e) => <h4 key={e}>{e}</h4>)
    
    return (
        <>
            <div className="container">
                <div className="card">
                    <div className="card_front">
                        <img src={img} alt={project} />
                        <h4> {project} </h4>
                    </div>
                    <div className="card_back">
                        <div className="place">
                            <h3>Tecnolog√≠as:</h3>
                            <div className="tech">{result}</div>
                        </div>
                        <div className="card_info">
                            <a href={url} target="_blank" rel="noopener noreferrer" >Repo</a>
                            {demo && <a href={demo} target="_blank" rel="noopener noreferrer">Demo</a>}
                        </div>
                    </div>
                </div>
            </div>
        </>
    )
}
