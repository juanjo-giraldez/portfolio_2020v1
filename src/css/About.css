

h1 {
    
    text-align: center;   
}

img {
    max-width: 100%;
    width: 50%;
    height: 50%;
    
}

.tama√±o {
    font-size: 5rem;
}

.info {
    display:flex;
    margin: 3rem;
    justify-content: space-around;;
    align-items: center;;
}

.info_imagen {
    padding: 2rem;
}
.info_p {
    align-items: center;
    padding: 2rem;
}

@media screen and (max-width: 768px) {

    img {
        width: 100%;
        height: 100%;
    }

   
    .info {
        display: flex;
        flex-wrap: wrap;
    }

    .info_p {
        padding: .5rem;
    }

    .skill_tech {
        display: flex;
        flex-direction: column;
    }
    
}

.skill_tech {
    display:flex;
    margin: 3rem;
    justify-content: space-around;
    align-items: flex-start;;

}

ul {
    min-width: 300px;
}
li {
    position: relative;
    list-style: none;
    margin: 2rem 0;
    font-weight: 500;

}

li:before {
     content: '';
     position: absolute;
     top: calc(100% + 10px);
     left: 0;
     width: 100%;
     height: 15px;
     background-color: rgba(0, 0, 0, .5);
     border-radius: 1000px;

}

li:after {
    content: '';
    position: absolute;
    top: calc(100% + 10px);
    left: 0;
    width: 100%;
    height: 15px;
    background-color: rgba(0, 0, 0, 1);
    border-radius: 1000px;
    animation-duration: 2s;
    animation-timing-function: ease;
    animation-fill-mode: forwards;


}
li.javascript:after {
    animation-name: javascript;
}

@keyframes javascript {
    from {
        width: 0%;
    }
    to {
        width: 80%;
    }
}

li.html:after {
    animation-name: html;
}

@keyframes html {
    from {
        width: 0%;
    }
    to {
        width: 80%;
    }
}


li.css3:after {
    animation-name: css3;
}

@keyframes css3 {
    from {
        width: 0%;
    }
    to {
        width: 70%;
    }
}

li.sass:after {
    animation-name: sass;
}

@keyframes sass {
    from {
        width: 0%;
    }
    to {
        width: 60%;
    }
}

li.react:after {
    animation-name: react;
}

@keyframes react {
    from {
        width: 0%;
    }
    to {
        width: 75%;
    }
}

li.angular:after {
    animation-name: angular;
}

@keyframes angular {
    from {
        width: 0%;
    }
    to {
        width: 70%;
    }
}

li.vue:after {
    animation-name: vue;
}

@keyframes vue {
    from {
        width: 0%;
    }
    to {
        width: 65%;
    }
}

li.bootstrap:after {
    animation-name: bootstrap;
}

@keyframes bootstrap {
    from {
        width: 0%;
    }
    to {
        width: 60%;
    }
}

li.prime_vue:after {
    animation-name: prime_vue;
}

@keyframes prime_vue {
    from {
        width: 0%;
    }
    to {
        width: 60%;
    }
}

li.prime_ng:after {
    animation-name: prime_ng;
}

@keyframes prime_ng {
    from {
        width: 0%;
    }
    to {
        width: 60%;
    }
}

li.mongo_db:after {
    animation-name: mongo_db;
}

@keyframes mongo_db {
    from {
        width: 0%;
    }
    to {
        width: 70%;
    }
}


li.mongoose:after {
    animation-name: mongoose;
}

@keyframes mongoose {
    from {
        width: 0%;
    }
    to {
        width: 65%;
    }
}

li.apis:after {
    animation-name: apis;
}

@keyframes apis {
    from {
        width: 0%;
    }
    to {
        width: 70%;
    }
}

li.node:after {
    animation-name: node;
}

@keyframes node {
    from {
        width: 0%;
    }
    to {
        width: 75%;
    }
}

li.express:after {
    animation-name: express;
}

@keyframes express {
    from {
        width: 0%;
    }
    to {
        width: 70%;
    }
}

li.npm:after {
    animation-name: npm;
}

@keyframes npm {
    from {
        width: 0%;
    }
    to {
        width: 65%;
    }
}

li.git:after {
    animation-name: git;
}

@keyframes git {
    from {
        width: 0%;
    }
    to {
        width: 70%;
    }
}

li.docker:after {
    animation-name: docker;
}

@keyframes docker {
    from {
        width: 0%;
    }
    to {
        width: 65%;
    }
}
