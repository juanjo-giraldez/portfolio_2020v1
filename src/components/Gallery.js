import React from 'react';
import { useProjects } from '../hooks/useProjects';
import { ProjectItem } from './ProjectItem';

export const Gallery = () => {

    const { data } = useProjects()

    return (
        <div>
            <h1>Estos   son mis proyectos</h1>
            <div className="card-grid">
                {data.map((e) =>
                    (< ProjectItem key={e.id} {...e} />))}
            </div>
        </div>
    )
}