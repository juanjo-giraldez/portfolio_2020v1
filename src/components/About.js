import React from 'react';
import '../css/About.css';
import Background from '../img/Background.jpg';

export const About = () => {



    return (
        <div>
            <div>
                <h1 className="tamaño">Acerca de mí</h1>
            </div>
            <div className="info">
                <img className="info_image" src={Background} alt="imagen" />
                <div  className="contain_p">
                    <p>
                        Mi interés por aprender me llevo hasta al sector del Web Developing.
                        donde me inicie. Me propuse conocer las TICs , lanzandome a realizar
                        un Bootcamp en Desarrollo web y poseer experiencia en proyectos.
                       </p>
                    <p>Dispuesto a emprender nuevas aventuras por descubrir.....¿me
                    acompañas? .
                     </p>
                    <a
                        href="https://drive.google.com/file/d/1L2V9ZH55eLbw0I-cqoM5sPwS-Kci8ISi/view?usp=sharing"
                        target="_blank" rel="noopener noreferrer"
                    >resume</a>
                </div>
            </div>


        </div >

    )
}
