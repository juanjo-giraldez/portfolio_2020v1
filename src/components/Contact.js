import React from 'react';
import '../css/contact.css';

export const Contact = () => {
    return (
        <>


            <h1 className="title">Contacta conmigo</h1>

            <table className="contact">
                <tbody>
                    <tr>
                        <td>
                            <p>Email:</p>
                        </td>
                        <th><a href='mailto:jjgiraldezjurado@gmail.com'>jjgiraldezjurado@gmail.com</a></th>
                    </tr>
                    <tr>
                        <td>
                            <p>LinkedIn:</p>
                        </td>
                        <th><a href='https://www.linkedin.com/in/juanjogirÃ¡ldez'
                            target="_blank" rel="noopener noreferrer">juanjo-giraldez</a></th>
                    </tr>
                    <tr>
                        <td>
                            <p>Github:</p>
                        </td>
                        <th><a href='https://github.com/juanjo-giraldez'
                            target="_blank" rel="noopener noreferrer">juanjo-giraldez</a></th>
                    </tr>
                    <tr>
                        <td>
                            <p>Gitlab:</p>
                        </td>
                        <th><a href='https://gitlab.com/juanjogiraldez_kairosds'
                            target="_blank" rel="noopener noreferrer">juanjogiraldez_kairosds</a></th>
                    </tr>
                </tbody>
            </table>

        </>
    )
}


