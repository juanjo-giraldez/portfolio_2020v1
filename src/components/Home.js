import React from 'react';
import '../css/home.css';

export const Home = () => {
    return (
        <div className="hero">
            <div className="contenedor">
                <div className="contenido">
                    <h2 className="subtitulo">Hola,  Soy </h2>
                    <h1 className="titulo">Juanjo Gir√°ldez
                    </h1>

                </div>
            </div>

        </div>
    )
}
