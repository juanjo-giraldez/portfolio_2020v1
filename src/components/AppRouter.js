import React from 'react';
import { BrowserRouter as Router, Redirect, Route, Switch } from "react-router-dom";
import { About } from "./About";
import { Gallery } from "./Gallery";
import { Home } from "./Home";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
    return (
        <Router >
        <div>
            <NavBar />
            <div>

                <Switch>
                    <Route exact path="/" component={ Home } />
                    <Route exact path="/about" component={ About } />
                    <Route exact path="/proyects" component={ Gallery } />
                    <Redirect to="/" />
                </Switch>

            </div>
        </div>

    </Router>
    )
}
