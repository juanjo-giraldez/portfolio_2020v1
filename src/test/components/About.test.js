import { shallow } from 'enzyme';
import React from 'react';
import { About } from "../../components/About";

describe('Test in <About />', () => {

    it('should show success Web Component', () => {

        const wrapper =  shallow(<About />);

        expect(wrapper).toMatchSnapshot();
        
    })
    
    
})



