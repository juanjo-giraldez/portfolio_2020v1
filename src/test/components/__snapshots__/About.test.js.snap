// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <About /> should show success Web Component 1`] = `
<div>
  <div>
    <h1
      className="tamaño"
    >
      Acerca de mí
    </h1>
  </div>
  <div
    className="info"
  >
    <img
      alt="imagen"
      className="info_image"
      src="Background.jpg"
    />
    <div
      className="contain_p"
    >
      <p>
        Entusiasta por el universo de la tecnología y por los nuevos avances; me propuse formarme en Web Development realizando un Bootcamp.
      </p>
      <p>
        Mi fin es alcanzar algún proyecto que me permita seguir avanzando en mis conocimientos en desarrollo y ponerlos en práctica.
      </p>
      <p>
        Mejorar a traves de desafios y progresar.
      </p>
      <p>
         Dispuesto a descubrir nuevos retos  para crecer .....¿me acompañas a emprender esta aventura? .
      </p>
      <a
        href="https://drive.google.com/file/d/1L2V9ZH55eLbw0I-cqoM5sPwS-Kci8ISi/view?usp=sharing"
        rel="noopener noreferrer"
        target="_blank"
      >
        resume
      </a>
    </div>
  </div>
</div>
`;
